lib/context.js

```diff
@@ -66,27 +65,22 @@ api.process = async ({
     return activeCtx;
   }
 
-  // track the previous context
-  const previousContext = activeCtx.previousContext || activeCtx;
-
-  // if context is property scoped and there's a previous context, amend it,
-  // not the current one
-  if(isPropertyTermScopedContext && activeCtx.previousContext) {
-    // TODO: consider optimizing to a shallow copy
-    activeCtx = activeCtx.clone();
-    activeCtx.isPropertyTermScoped = true;
-    activeCtx.previousContext = await api.process({
-      activeCtx: activeCtx.previousContext,
-      localCtx: ctxs,
-      options,
-      isPropertyTermScopedContext
-    });
-    return activeCtx;
+  // override propagate if localCtx has `@propagate`
+  if(_isObject(ctxs[0]) && '@propagate' in ctxs[0]) {
```

Comment:
I see "error checking done later" but perhaps this should be doing:

```suggestion
  if(_isObject(ctxs[0]) && typeof ctxs[0]['@propagate'] === 'boolean') {
```